aea/test_tools/test_cases.py

```diff
@@ -766,6 +768,11 @@ def change_directory(cls, path: Path) -> None:
     @classmethod
     def _terminate_subprocesses(cls) -> None:
         """Terminate all launched subprocesses."""
+        warnings.warn(
+            "BaseAEATestCase._terminate_subprocesses is deprecated and will be removed next release",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
this is only called in `teardown_class` and unnecessary, as we will now raise a `subprocess.TimeoutExpired` in case we cannot terminate a subprocess in our `BaseAEATestCase`.

The logic also made little sense. We already invoked subprocess.kill() before reaching this in the original implementation.